app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -360,4 +360,42 @@ object BitcoindRpcBackendUtil extends Logging {
 
     resultF
   }
+
+  def startBitcoindMempoolPolling(
+      bitcoind: BitcoindRpcClient,
+      interval: FiniteDuration = 10.seconds)(
+      processTx: Transaction => Future[Unit])(implicit
+      system: ActorSystem,
+      ec: ExecutionContext): Cancellable = {
+    @volatile var prevMempool: Set[DoubleSha256DigestBE] =
+      Set.empty[DoubleSha256DigestBE]
+
+    def getDiffAndReplace(
+        newMempool: Set[DoubleSha256DigestBE]): Set[DoubleSha256DigestBE] = {
+      val txids =
+        if (prevMempool.isEmpty) Set.empty[DoubleSha256DigestBE]
+        else
+          newMempool.diff(prevMempool)
```

Comment:
won't this constantly re-add txs to the mempool? I.e.

1. We run this scheduled job and find `tx1` in the mempool. `prevMempool` is empty, we add `tx1`.
2. After 10 seconds we run this scheduled job and find `tx2` in the mempool. We evict `tx1` from `prevMempool` since we have already seen it.
3. After another 10 seconds, we run this scheduled job and find `tx1` _again_ in the mempool. A block has not ocurred. Since it's not in `prevMempool` we add it again. We remove `tx2`. 

We do this over and over again until a block is found.

I don't know how to get around this, unless we just remove txs from the mempool when a block is found :/